dynamicconfig: Enables dynamic-config and drift-checker by default

ovidiutirla opened the pull request:
tl;dr
The agent will monitor the designated ConfigMap and automatically synchronize any changes with its in-memory database, StateDB. Create a new cell in the Agent that watches the configured ConfigMap and creates a StateDB table to store the key-value pairs. The Agent cell will expose the read-only version of the StateDB table.
The monitored sources are configured by the initContainer at start-up, see https://github.com/cilium/cilium/blob/236bc2eac4c36cff60e0079f06d33816b3344ccc/pkg/option/resolver/resolver.go#L110

--
**Drift Checker** will compare the DynamicConfig Table content with the Agent config and compute the drift delta, logging the deltas and publish relevant metrics. This functionality allows the cluster administrators to easily track config drifts.
Publishes metrics counting the delta length and labeling by the checksum.

<details>

<summary> Grafana metrics</summary>
Agent start -> Drift appears -> delete all agent pods
<img width="1986" alt="image" src="https://github.com/user-attachments/assets/4ea2886a-5b62-4e1c-9a24-8828122b2784">

Start -> Drift -> kubectl rollout restart daemonset cilium -n kube-system

<img width="1984" alt="image" src="https://github.com/user-attachments/assets/b49a4d8f-52d5-4eda-8db9-bb86e4078a14">

</details>

Related: https://github.com/cilium/cilium/issues/27972, https://github.com/cilium/cilium/issues/34101

```release-note
Introduces cilium-config ConfigMap monitoring, allowing the agent to automatically synchronize changes in designated sources with its in-memory database. A new drift checker compares the ingested configurations with the agent's configuration, logging any differences and publishing metrics for easy tracking of configuration drifts. 
```

ovidiutirla committed:
dynamicconfig: Enable dynamic config and drift checker by default

Signed-off-by: Ovidiu Tirla <otirla@google.com>

ovidiutirla committed:
docs: Enable dynamic config and drift checker by default

Signed-off-by: Ovidiu Tirla <otirla@google.com>

bimmlerd reviewed (changes requested):
Hi- code changes are trivial and are ok, but the implications aren't clear to me yet
Can you motivate a bit more why this should be enabled by default? Has it seen prod use already? I'm asking because turning it on seems to cause a related panic in tests https://github.com/cilium/cilium/actions/runs/12275147175/job/34249594941#step:13:894 so I'm not sure this is ready.

Re 1.17 - I believe the feature freeze is in effect until 1.17 is branched, so I'm not sure this can still go in unfortunately.

ovidiutirla committed:
test: Add ConfgMapList to selector filtering

Signed-off-by: Ovidiu Tirla <otirla@google.com>

ovidiutirla committed:
test: Allow driftchecker to exit when context is done

Some tests have no config populated so the table is empty and the tests are stuck waiting to get elements in the table.

Signed-off-by: Ovidiu Tirla <otirla@google.com>
